### PR TITLE
Add location prompt handling for IAM clicks

### PR DESCRIPTION
**Pull request by Jeasmine**
* Add location prompt to IAM action
   * Show location prompt when available
   * Show multiple prompts when available
   * Add test for prompt and multiple prompts
   * Pause IAM display if user prompt displayed
   * Wait for app to be active for IAM evaluation

---
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-ios-sdk/629)

### CODE REVIEW

File: iOS_SDK/OneSignalSDK/Source/OneSignalLocation.m [hunk 1]
```diff
@@ -218,8 +250,10 @@ + (void)internalGetLocation:(bool)prompt {
 
 - (void)locationManager:(id)manager didUpdateLocations:(NSArray *)locations {
```
Comment (jkasten2) [inline iOS_SDK/OneSignalSDK/Source/OneSignalLocation.m]: Would this method still fire if the user has accepted notifications but has disabled location in the device settings?

File: iOS_SDK/OneSignalSDK/UnitTests/InAppMessagingIntegrationTests.m [hunk 2]
```diff
@@ -888,6 +888,51 @@ - (void)testIAMClickedLaunchesPrompt {
...
+    let registrationResponse = [OSInAppMessageTestHelper testRegistrationJsonWithMessages:@[message]];
+
+    // the trigger should immediately evaluate to true and should
+    // be shown once the SDK is fully initialized.
+    [OneSignalClientOverrider setMockResponseForRequest:NSStringFromClass([OSRequestRegisterUser class]) withResponse:registrationResponse];
+
+    [UnitTestCommonMethods initOneSignalAndThreadWait];
+
+    NSMutableDictionary *actionJson = [OSInAppMessageTestHelper.testActionJson mutableCopy];
+    [actionJson setValue:@[@"location"] forKey:@"prompts"];
```
Comment (jkasten2) [inline iOS_SDK/OneSignalSDK/UnitTests/InAppMessagingIntegrationTests.m]: nit, `actionJson[@"prompts"] = @"location"`  is easier to read and more consistent with other dictionary setting we have

File: iOS_SDK/OneSignalSDK/Source/OSMessagingController.m [hunk 3]
```diff
@@ -200,7 +203,12 @@ - (void)presentInAppMessage:(OSInAppMessage *)message {
         // Return early if an IAM is already showing
         if (self.isInAppMessageShowing)
             return;
-        
+        // Return early if the app is not active
+        if ([[UIApplication sharedApplication] applicationState] != UIApplicationStateActive) {
+            [OneSignal onesignal_Log:ONE_S_LL_VERBOSE message:@"Pause IAMs display due to app inactivity"];
+            _pausedByInactive = YES;
```
Comment (mikechoch) [inline iOS_SDK/OneSignalSDK/Source/OSMessagingController.m]: Change name of `_pausedByInactive` to `_isAppInactive`

File: iOS_SDK/OneSignalSDK/Source/OneSignalLocation.m [hunk 4]
```diff
@@ -108,8 +118,7 @@ + (void)getLocation:(bool)prompt {
             [OneSignalLocation internalGetLocation:prompt];
         });
     }
-    
-    //Listen to app going to and from background
+    // Listen to app going to and from background
 }
 
 + (void)onfocus:(BOOL)isActive {
```
Comment (mikechoch) [inline iOS_SDK/OneSignalSDK/Source/OneSignalLocation.m]: `onFocus`

File: iOS_SDK/OneSignalSDK/Source/OneSignalLocation.m (hunk 1 above)
Comment (mikechoch) [reply iOS_SDK/OneSignalSDK/Source/OneSignalLocation.m]: Good question

File: iOS_SDK/OneSignalSDK/UnitTests/InAppMessagingIntegrationTests.m [hunk 5]
```diff
@@ -888,6 +888,51 @@ - (void)testIAMClickedLaunchesPrompt {
...
+    [OneSignalClientOverrider setMockResponseForRequest:NSStringFromClass([OSRequestRegisterUser class]) withResponse:registrationResponse];
+
+    [UnitTestCommonMethods initOneSignalAndThreadWait];
+
+    NSMutableDictionary *actionJson = [OSInAppMessageTestHelper.testActionJson mutableCopy];
+    [actionJson setValue:@[@"push", @"location"] forKey:@"prompts"];
+    let action = [OSInAppMessageAction instanceWithJson: actionJson];
+    let testMessage = [OSInAppMessage instanceWithJson:message];
+    
+    XCTAssertFalse(action.promptActions[0].didAppear);
```
Comment (mikechoch) [inline iOS_SDK/OneSignalSDK/UnitTests/InAppMessagingIntegrationTests.m]: `didAppear` rename to `hasPrompted`

File: iOS_SDK/OneSignalSDK/Source/OneSignalLocation.m (hunk 1 above)
Comment (Jeasmine) [reply iOS_SDK/OneSignalSDK/Source/OneSignalLocation.m]: Yes! Discussed on the call 😁

File: iOS_SDK/OneSignalSDK/UnitTests/Shadows/OSMessagingControllerOverrider.m [hunk 6]
```diff
@@ -94,7 +94,7 @@ + (void)dismissCurrentMessage {
 }
 
 + (BOOL)currentPromptAppear {
```
Comment (mikechoch) [inline iOS_SDK/OneSignalSDK/UnitTests/Shadows/OSMessagingControllerOverrider.m]: Rename to `isPromptShowing`

Comment (Jeasmine) [reply iOS_SDK/OneSignalSDK/UnitTests/Shadows/OSMessagingControllerOverrider.m]: I'm not using this anymore, good catch

**Pull request merged**